src/test/java/org/ow2/proactive/procci/service/occi/InstanceServiceTest.java

```diff
@@ -0,0 +1,225 @@
+
+    }
+
+    @Test
+    public void getMockedEntityTest(){
+
+        String id = "idTest";
+        Model model = new Model.Builder("modelTest","actionTest")
+                .addVariable(MetamodelAttributes.ID_NAME,id)
+                .build();
+
+
+        when(cloudAutomationInstanceClient.getInstanceByVariable(ID_NAME, ConvertUtils.formatURL(id)))
+                .thenReturn(Optional.of(model));
+
+        Optional<Entity> entity = instanceService.getEntity(id);
+
+        assertThat(entity.isPresent()).isTrue();
+        assertThat(entity.get().getId()).matches(id);
+
+        String id2 = "idTest2";
+        Model model2 = new Model.Builder(BigDataIdentifiers.SWARM_MODEL,"actionTest")
+                .addVariable(MetamodelAttributes.ID_NAME,id)
+                .addVariable(BigDataAttributes.HOST_IP_NAME,"hostTest")
+                .addVariable(BigDataAttributes.MASTER_IP_NAME,"masterTest")
+                .build();
+
+
+        when(cloudAutomationInstanceClient.getInstanceByVariable(ID_NAME, ConvertUtils.formatURL(id2)))
+                .thenReturn(Optional.of(model2));
+        when(mixinService.getEntityMixinNames(id2))
+                .thenReturn(new HashSet<>());
+
+        Optional<Entity> entity2 = instanceService.getEntity(id2);
+
+        assertThat(entity2.isPresent()).isTrue();
+        assertThat(entity2.get().getId()).matches(id);
+        assertThat(entity2.get()).isInstanceOf(Resource.class);
+        assertThat(entity2.get()).isInstanceOf(Swarm.class);
+    }
+
+    @Test
+    public void getInstancesRenderingTest(){
+        Model compute = new Model
+                .Builder(InfrastructureIdentifiers.COMPUTE_MODEL,"action")
+                .addVariable(MetamodelAttributes.ID_NAME,"id1")
+                .build();
+
+        JSONObject jsonObject = new JSONObject();
+        jsonObject.put("id1",compute.getJson());
+
+        when(cloudAutomationInstanceClient.getRequest())
+                .thenReturn(jsonObject);
+
+        when(mixinService.getEntityMixinNames("id1"))
+                .thenReturn(new HashSet<>());
+
+        List<EntityRendering> renderings = instanceService.getInstancesRendering();
+
+
+        assertThat(renderings.get(0).getKind())
+                .matches(InfrastructureIdentifiers.INFRASTRUCTURE_SCHEME+InfrastructureIdentifiers.COMPUTE);
+
+        Model resource = new Model.Builder("test","action")
+                .build();
+
+        jsonObject = new JSONObject();
+        jsonObject.put("id2",resource.getJson());
+
+        when(cloudAutomationInstanceClient.getRequest())
+                .thenReturn(jsonObject);
+
+        when(mixinService.getEntityMixinNames("id2"))
+                .thenReturn(new HashSet<>());
+
+        renderings = instanceService.getInstancesRendering();
+
+        assertThat(renderings.get(0).getKind())
+                .matches(MetamodelIdentifiers.CORE_SCHEME+MetamodelIdentifiers.RESOURCE_TERM);
+    }
+
+    @Test
+    public void createTest(){
```

Comment:
:+1: For test cases. When you create test cases you can give them super long names like "testThatAllInstancesStayTheSameFromObjectCreationTimeAndGettersReturnThe SameInstance" ..... 

Looks meh but the reader knows exactly what you tested there (because you explain it in a sentence). That also helps you to keep tests small. When you test a very precise defined point and the test cases has such a long name. Then when the tests fails, then we can see very fast what is wrong and are probably faster in fixing it.